universal-application-tool-0.0.1/app/services/applicant/question/FileUploadQuestion.java

```diff
@@ -63,22 +68,24 @@ public ValidationErrorMessage fileRequiredMessage() {
   }
 
   public Optional<String> getFileKeyValue() {
-    if (fileKeyValue != null) {
-      return fileKeyValue;
+    if (fileKeyValue.get().isPresent()) {
```

Comment:
remove the get here as you're seeing if the cached value has been set yet. The code here is assuming it is (which should [throw an exception](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#get()) on the first call also, I'd expect some test to fail)

Adding Cache to this name I think will help too

```
if (fileKeyValueCache.isPresent()) {
  return fileKeyValueCache.get();
}
```

Reply:
yep! looks good.